### ISSUE TITLE
Support for graphite pickle output

### ISSUE THREAD

**Issue by sie-dlanz** [influxdata/telegraf#6575]
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal:

Implement graphite's pickle output protocol support

### Current behavior:

Telegraf's graphite output plugin only supports plaintext (default TCP 2003) communications

### Desired behavior:

Support graphite's pickled protocol (default TCP 2004)

### Use case:

Graphite's pickled communications are much faster and more efficient for a large number of metric datapoints being submitted in one go, allowing batched submissions.

**Comment by danielnelson** [comment]
Good idea, we could maybe use https://github.com/hydrogen18/stalecucumber, though I wonder if we can get away with a much simpler and targetted implementation of only what we need.

**Comment by sie-dlanz** [comment]
There's a helper library specifically targeted at graphite here https://github.com/lomik/graphite-pickle and the carbon-relay-ng package is written in golang and has the pickle implementation here https://github.com/graphite-ng/carbon-relay-ng/blob/master/input/pickle.go maybe this can be of somewhat direct use?

**Comment by sspaink** [comment]
@sie-dlanz are you still interested in this feature? It seems https://github.com/hydrogen18/stalecucumber and https://github.com/lomik/graphite-pickle both haven't been updated in a long time which makes me hesitant to use them in Telergaf. Using the carbon-relay implementation seems like a good idea, but also hasn't been touched in 4 years so curious if this is still an actively used protocol?

**Comment by srebhan** [comment]
Is there anyone still interested in this feature?

**Comment by Hipska** [comment]
I think I would be interested. Luckily my endpoint also supports JSON, which is also better/compacter than regular graphite.

**Comment by srebhan** [comment]
@Hipska do you know of a working golang implementation of that format?

**Comment by Hipska** [comment]
As said before, the one from @lomik would be my first place to look. The reason, there aren't much updates is that it is stable..
